api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -102,7 +103,7 @@ public Schema(int schemaId, List<NestedField> columns, Map<String, Integer> alia
 
     this.identifierFieldIds = identifierFieldIds != null ? Ints.toArray(identifierFieldIds) : new int[0];
 
-    lazyIdToName();
```

Comment:
The purpose of this call is to ensure that `lazyIdToName()` is called to index the schema. That will fail if there are duplicate IDs. Since that's what we use to find the highest field ID, this line now does both.

Reply:
Oops, I did not see a call to `lazyIdToName()`.